File: leetcode-solved/3172-divisible-and-non-divisible-sums-difference/solution.py
class Solution:
    def differenceOfSums(self, n: int, m: int) -> int:
        sumAll = (1+n)*n//2
        sumMultiples = sum(num for num in range(m,n+1,m))
        sumNonMultiples = sumAll - sumMultiples
        return sumNonMultiples - sumMultiples
        

File: leetcode-solved/3172-divisible-and-non-divisible-sums-difference/solution.java
class Solution {
    public int differenceOfSums(int n, int m) {
        int sumAll = (1+n)*n/2;
        int sum2 = 0;
        if(n>=m){
            for(int num=m;num<=n; num+=m){
                sum2 += num;
            }
        } 
        int sum1 = sumAll - sum2;
        return sum1 - sum2;
    }
}
